docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,562 @@
+# Diamond workflow support
+
+A diamond workflow allows for a pipeline to scale out the generation of a single bundle.
```

Comment:
Should we close https://github.com/oneconcern/datamon/pull/399

Reply:
I made (and sync'ed) all these PRs to produce a _reviewable_ ensemble.
I'd prefer that we merge them one by one following informed and understood agreement.